core/src/test/scala/unit/kafka/server/DynamicConfigChangeTest.scala

```diff
@@ -65,8 +66,8 @@ class DynamicConfigChangeTest extends KafkaServerTestHarness {
 
     val quotaManagers = servers.head.apis.quotas
     rootEntityType match {
-      case ConfigType.Client => AdminUtils.changeClientIdConfig(zkUtils, configEntityName, props)
-      case _ => AdminUtils.changeUserOrUserClientIdConfig(zkUtils, configEntityName, props)
+      case ConfigType.Client => AdminUtils.changeClientIdConfig(zkUtils, Sanitizer.sanitize(configEntityName), props)
+      case _ => AdminUtils.changeUserOrUserClientIdConfig(zkUtils, Sanitizer.sanitize(configEntityName), props)
     }
```

Comment:
The sanitization needs to be done by the caller so that just the names that need sanitizing get sanitized.